### PR TITLE
Added Sections and a Button Change

### PR DESCRIPTION
**Pull request by isaacruz2308**
Divided the orders by status section and changed delete functionality to reject instead

### CODE REVIEW

File: frontend/src/modules/staff/components/orders/OrderDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const OrderDialog: React.FC<OrderDialogProps> = ({
+  open,
+  handleCloseDialog,
+  confirmRemoveOrReject,
+}) => {
+  return (
+    <Dialog
+      open={open}
```
Comment (isabellagarrido) [inline frontend/src/modules/staff/components/orders/OrderDialog.tsx]: Can we use the reusable [baseDialog](https://github.com/UPRMeals/UPRMeals/blob/main/frontend/src/shared/components/baseDialog.tsx) here? There's a couple of examples in the code, puedes buscar ejemplos en el `modules/staff/components` folder: https://github.com/UPRMeals/UPRMeals/tree/main/frontend/src/modules/staff/components.

File: frontend/src/modules/staff/components/orders/OrderList.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            <Typography
+              variant="h6"
+              sx={{ fontWeight: "medium", color: "primary.main", mb: 1 }}
+            >
+              {status}
+            </Typography>
+            <Divider
+              variant="middle"
+              sx={{ mb: 2, boxShadow: "0px 2px 4px rgba(0,0,0,0.1)" }}
+            />
```
Comment (isabellagarrido) [inline frontend/src/modules/staff/components/orders/OrderList.tsx]: What would you think about removing the divider altogether?

File: frontend/src/modules/staff/components/orders/OrderCard.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+const OrderCard: React.FC<Props> = ({
+  order,
+  expanded,
+  handleExpandClick,
+  handleStatusChange,
+  handleRemoveOrder,
+}) => {
+  return (
+    <Card variant="outlined" sx={{ m: 1, boxShadow: 3, overflow: "visible" }}>
```
Comment (isabellagarrido) [inline frontend/src/modules/staff/components/orders/OrderCard.tsx]: What do you think of using this box shadow? We've been using it in other parts of the app.

```suggestion
    <Card variant="outlined" sx={{ m: 1, boxShadow: "1px 2px 7px 1px " + blueGrey[100] + "99", overflow: "visible" }}>
```

File: frontend/src/pages/staff/orders/index.tsx [hunk 4]
```diff
@@ -1,222 +1,132 @@
...
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleCloseDialog} color="primary">
+            Cancel
+          </Button>
+          <Button onClick={confirmRemoveOrReject} color="primary" autoFocus>
+            Confirm
+          </Button>
+        </DialogActions>
+      </Dialog>
```
Comment (isabellagarrido) [inline frontend/src/pages/staff/orders/index.tsx]: I think you created the OrderDialog but forgot to use it here lol

**Review by isabellagarrido — COMMENTED** [review]
It looks really nice! just had a couple of comments :)

File: frontend/src/modules/staff/components/orders/OrderCard.tsx (hunk 3 above)
Comment (isaacruz2308) [reply frontend/src/modules/staff/components/orders/OrderCard.tsx]: looks super nice

File: frontend/src/pages/staff/orders/index.tsx (hunk 4 above)
Comment (isaacruz2308) [reply frontend/src/pages/staff/orders/index.tsx]: Yep! Estaba moviendo todo a los modules pero en el apuro se me olvido esa jajajaj

Thanks for pointing it out

File: frontend/src/modules/staff/components/orders/OrderCard.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        <CardContent>
+          <Typography paragraph sx={{ pb: 0, mb: 0, fontWeight: "bold" }}>
+            Order Items:
+          </Typography>
+          <List dense sx={{ pl: 2 }}>
+            {order.orderItems.map((orderItem, index) => (
+              <ListItem key={index}>
+                <ListItemText
+                  primary={orderItem.item.name}
+                  //secondary={`Quantity: ... (however you calculate quantity)`}
```
Comment (Kenneth-Aponte) [inline frontend/src/modules/staff/components/orders/OrderCard.tsx:115]: Is this still pending?

Comment (isaacruz2308) [reply frontend/src/modules/staff/components/orders/OrderCard.tsx:115]: I was going to ask about this. Right now, if an item is duplicated I just display it twice. Dont know if I should calculate duplicates and add that field tbh. Wdyt?

Comment (isabellagarrido) [reply frontend/src/modules/staff/components/orders/OrderCard.tsx:115]: I think displaying it twice is ok.

Comment (Kenneth-Aponte) [reply frontend/src/modules/staff/components/orders/OrderCard.tsx:115]: I agree with isabella, in the order status page its also displayed twice.

File: backend/src/order/order.service.ts [hunk 6]
```diff
@@ -5,13 +5,15 @@ import {
...
-import { Item, ItemType } from '@prisma/client';
+import { Item, ItemType, OrderStatusType } from '@prisma/client';
+import { endOfDay, startOfDay } from 'date-fns';
 
 @Injectable()
 export class OrderService {
   constructor(private prismaService: PrismaService) {}
 
   async createOrder(userId: number, data: CreateOrderData): Promise<any> {
+    if (!userId) throw new Error('User not found');
```
Comment (Kenneth-Aponte) [inline backend/src/order/order.service.ts:16]: Seems like this line is duplicated.

File: frontend/src/modules/staff/components/orders/types.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  orderCombos: OrderCombo[];
+  createdAt: Date;
+  updatedAt: Date;
+  removed: boolean;
+}
+
+export const statusColors: Record<
+  OrderStatus,
+  "default" | "primary" | "error" | "success" | "warning"
+> = {
```
Comment (Kenneth-Aponte) [inline frontend/src/modules/staff/components/orders/types.ts:72]: On the order status page on the user, there is a [list of colors](https://github.com/UPRMeals/UPRMeals/blob/836eafda5fd095e939f488d684ea7995e75c4208/frontend/src/pages/customers/orders/index.tsx#L20), what do you think of those? Feel free to leave these as is though. Also if you suggest that these are a better fit then we can refactor those as well for consistency :)

File: backend/src/order/order.service.ts (hunk 6 above)
Comment (isaacruz2308) [reply backend/src/order/order.service.ts:16]: thanks for pointing it out

File: frontend/src/modules/staff/components/orders/types.ts (hunk 7 above)
Comment (isaacruz2308) [reply frontend/src/modules/staff/components/orders/types.ts:72]: No problem at all. Done

**Review by Kenneth-Aponte — APPROVED** [review]

**Pull request merged**